### DISCUSSION TITLE
ChatGPT dynamic body

### DISCUSSION THREAD

**Discussion by valepetros** [LiveHelperChat/livehelperchat#2135] [Q&A]
Hello, I've followed the tutorial to use chatgpt in LHC and it works well.

I'm working on a php script that generate the json request for ChatGPT, there's a way to push the output into the rest api body panel of Live helper chat?

thanks

**Comment by remdex** [comment]
You can listen to this event and override rest api params

'''erLhcoreClassChatEventDispatcher::getInstance()->dispatch('chat.rest_api_before_request', array(
                    'restapi' => & $restAPI,
                    'chat' => $chat,
                    'params' => $params,
                    'method' => & $method
                ));
'''

**Reply by valepetros** [reply]
can you better explain how can I use this override script, please?

**Comment by valepetros** [comment]
how can I override this event? 
Can you provide me an example of valid data for $restapi variable?

thanks
